xbmc/listproviders/StaticProvider.cpp

```diff
@@ -117,6 +117,6 @@ bool CStaticListProvider::AlwaysFocusDefaultItem() const
 
 bool CStaticListProvider::OnClick(const CGUIListItemPtr &item)
 {
-  CGUIStaticItemPtr staticItem = std::static_pointer_cast<CGUIStaticItem>(item);
+  CGUIStaticItem *staticItem = static_cast<CGUIStaticItem*>(item.get());
```

Comment:
Something missing in this commit? This changes nothing

Reply:
The problem is `item`'s lifetime ends with the final call in this function. Acquiring a shared pointer means that item is used after `item`'s lifetime ends.